src/components/estimationScore/services/estimatedScoreEventsHandlerSrv.js

```diff
@@ -0,0 +1,175 @@
+                    }
+
+                    var questionResults = exerciseResult.questionResults;
+
+                    var rawPoints = {
+                        total: questionResults.length * exercisesRawScoring[exerciseType].correctWithin,
+                        earned: 0
+                    };
+
+                    var allowedTimeForExercise = angular.isDefined(allowedTime) ? allowedTime : allowedTimeForExercisesMap[exerciseType];
+                    if(angular.isUndefined(allowedTimeForExercise)){
+                        $log.error('EstimatedScoreEventsHandlerSrv: allowed time missing for the following exercise type: ' + exerciseType);
+                    }
+                    var withinAllowedTime = allowedTimeForExercise >= exerciseResult.duration;
+                    questionResults.forEach(function (result) {
+                        rawPoints.earned += _getQuestionRawPoints(exerciseType, result, withinAllowedTime);
+                    });
+                    return rawPoints;
+                }
+
+                childScope.$on(exerciseEventsConst.section.FINISH, function (evt, section, sectionResult, exam) {
+                    var isDiagnostic = exam.typeId === ExamTypeEnum.DIAGNOSTIC.enum;
+                    if (isDiagnostic) {
+                        diagnosticSectionCompleteHandler(section, sectionResult);
+                    }
+                    var rawScore = calculateRawScore(ExerciseTypeEnum.SECTION.enum, sectionResult, section.time);
+                    EstimatedScoreSrv.addRawScore(rawScore, ExerciseTypeEnum.SECTION.enum, section.subjectId, section.id, isDiagnostic);
+                });
+
+                function _baseExerciseFinishHandler(exerciseType, evt, exercise, exerciseResult) {
+                    var rawScore = calculateRawScore(exerciseType, exerciseResult);
+                    EstimatedScoreSrv.addRawScore(rawScore, exerciseType, exercise.subjectId, exercise.id);
+                }
+
+                var exercisesHandledByBaseExerciseFinishHandler = [
+                    {
+                        name: exerciseEventsConst.drill.FINISH,
+                        type: ExerciseTypeEnum.DRILL.enum
+                    },
+                    {
+                        name: exerciseEventsConst.tutorial.FINISH,
+                        type: ExerciseTypeEnum.TUTORIAL.enum
+                    },
+                    {
+                        name: exerciseEventsConst.game.FINISH,
+                        type: ExerciseTypeEnum.GAME.enum
+                    }
+                ];
+                exercisesHandledByBaseExerciseFinishHandler.forEach(function (evt) {
+                    $rootScope.$on(evt.name, _baseExerciseFinishHandler.bind(EstimatedScoreEventsHandlerSrv, evt.type));
+                });
+
+                EstimatedScoreEventsHandlerSrv.init = angular.noop;
```

Comment:
This function is used in order to load the service in run time so it will start listening to events immediately as the app start.